### DISCUSSION TITLE
Check if directory exists.

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#288] [Q&A] [answered]
I came across an issue with my new panel that causes an error on autoload if the directory doesn't exist.

I found this little function to check.

```
function directoryExists(path)
  local f  = io.popen("cd " .. path)
  local ff = f:read("*all")

  if (ff:find("ItemNotFoundException")) then
    return false
  else
    return true
  end  
end
```

Does anyone know if there's a way to do this without io (ie using ctrlr's native file procedures)?  This works, but it flashes up a terminal window to do the check)

**Comment by dobo365** [comment, marked as answer]
For all your file handling, you should use the File class from the Juce library.
See: [https://docs.juce.com/master/classFile.html](https://docs.juce.com/master/classFile.html)

Example to load a file:
```
	f = utils.openFileWindow ("Load Pro 2 packed program dump", File(""), "*.syx", true)
	if f:isValid() then	-- A file has been selected and the user pressed Open
		if f:existsAsFile() then
```

Example to save a file:
```
	f = utils.saveFileWindow ("Save Pro 2 program buffer dump sysex file", File(""), "*.syx", true)
	if f:isValid() then
		if f:existsAsFile()==false then
			f:create()
		end
```

Finally, in your case, open the directory as a file object then test it with 
```
	f = File("xxxx")
 	if f:isDirectory() then
```

**Reply by Godlike-Productions** [reply]
Thanks so much, that's what I was looking for.
